### PR TITLE
🌱 lint: fix duplicated words via dupword linter

### PR DESCRIPTION
**Pull request by MrDXY**
**What this PR does / why we need it**:
Check the duplicate words in the code via linter configuration, and fix it. 
Enable `dupword` check in `.golangci.yml` configuration file.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of [#2058 ](url)

### CODE REVIEW

**Comment by zhanggbj** [comment]
/ok-to-test
And please sign the CLA following the guide above.

File: main.go [hunk 1]
```diff
@@ -254,7 +254,7 @@ func main() {
 	addToManager := func(ctx *context.ControllerManagerContext, mgr ctrlmgr.Manager) error {
 		tracker, err := setupRemoteClusterCacheTracker(ctx, mgr)
 		if err != nil {
-			return perrors.Wrapf(err, "unable to create remote cluster tracker tracker")
+			return perrors.Wrapf(err, "unable to create remote cluster tracker")
```
Comment (chrischdi) [inline main.go:257]:
```suggestion
			return perrors.Wrapf(err, "unable to create remote cluster cache tracker")
```

**Review by chrischdi — COMMENTED** [review]
one nit. 

Thanks for picking up this work!

File: main.go (hunk 1 above)
Comment (MrDXY) [reply main.go:257]: Got it!

**Review by chrischdi — COMMENTED** [review]
/lgtm

**Comment by MrDXY** [comment]
I apologize for any inconvenience. I've made several adjustments after your comment due to test failures attributed to `verify-gen` at `Makefile`. To elaborate, the problem arises from a diff inconsistency in the `config/default/crd/bases/infrastructure.cluster.x-k8s.io_vspherevms.yaml` file compared to the automatically generated CRD files. Consequently, I've modified the CRD file in our source code to ensure consistency.

**Comment by chrischdi** [comment]
> I apologize for any inconvenience. I've made several adjustments after your comment due to test failures attributed to `verify-gen` at `Makefile`. To elaborate, the problem arises from a diff inconsistency in the `config/default/crd/bases/infrastructure.cluster.x-k8s.io_vspherevms.yaml` file compared to the automatically generated CRD files. Consequently, I've modified the CRD file in our source code to ensure consistency.

I hope you did not do this manually :-) that's what is done via `make generate-manifests`.

/lgtm

**Comment by sbueringer** [comment]
> I apologize for any inconvenience.

No worries :). 

Thank you and welcome :)

/lgtm
/approve

**Pull request merged**